randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -1,12 +1,227 @@
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
+from randovania.exporter import pickup_exporter
+from randovania.exporter.hints import guaranteed_item_hint
 from randovania.exporter.patch_data_factory import BasePatchDataFactory
+from randovania.game_description.assignment import PickupTarget
+from randovania.games.am2r.exporter.hint_namer import AM2RHintNamer
+from randovania.games.am2r.layout.hint_configuration import ItemHintMode
 from randovania.games.game import RandovaniaGame
+from randovania.generator.pickup_pool import pickup_creator
+from randovania.lib import json_lib
+
+if TYPE_CHECKING:
+    from random import Random
 
 
 class AM2RPatchDataFactory(BasePatchDataFactory):
+    _EASTER_EGG_SHINY = 1024
+
+    def _does_pickup_have_shiny_values(self, pickup_list, item_effect: str, sprite_details: str, header: str,
+                                       for_other_player: bool, rng: Random) -> bool:
+        return (pickup_list["item_effect"] == item_effect and pickup_list["sprite_details"]["name"] == sprite_details
+                and pickup_list["text"]["header"] == header and not for_other_player and
+                rng.randint(0, self._EASTER_EGG_SHINY) == 0)
+
+    def _create_pickups_dict(self, pickup_list, item_info, rng: Random):
+        pickup_map_dict = {}
+        for pickup in pickup_list:
+            quantity = pickup.conditional_resources[0].resources[0][1]
+            object_name = self.game.region_list.node_from_pickup_index(pickup.index).extra["object_name"]
+            pickup_map_dict[object_name] = {
+                "sprite_details": {
+                    "name": item_info[pickup.model.name]["sprite_name"],
+                    "speed": item_info[pickup.model.name]["sprite_speed"]
+                },
+                "item_effect": pickup.original_pickup.name,
+                "quantity": quantity,
+                "text": {
+                    "header": item_info[pickup.name]["text_header"],
+                    "description": pickup.collection_text[0]
+                }
+            }
+            # Shiny Missiles
+            if (self._does_pickup_have_shiny_values(pickup_map_dict[object_name], "Missile Expansion", "sItemMissile",
+                                                    "Got Missile Tank", pickup.other_player, rng)):
+                pickup_map_dict[object_name]["sprite_details"]["name"] = "sItemShinyMissile"
+                pickup_map_dict[object_name]["text"]["header"] = "Got Shiny Missile Tank"
```

Comment:
```py
effect, sprite = sprite, header
shinies = {
    ("Missile Expansion", "sItemMissile"): ("sItemShinyMissile", "Got Shiny Missile Tank"),
    # etc
}
pickup_obj = pickup_map_dict[object_name]
shiny_id = (pickup_obj["item_effect"], pickup_obj["sprite_details"]["name"])
if (
    (shiny_id in shinies)
    and not pickup.other_player,
    and rng.randint(0, self._EASTER_EGG_SHINY) == 0
):
    sprite, text = shinies[shiny_id]
    pickup_obj["sprite_details"]["name"] = sprite
    pickup_obj["text"]["header"] = text
```
I'd suggest something like this, with a dict mapping all the shinies rather than duplicating code for every new shiny